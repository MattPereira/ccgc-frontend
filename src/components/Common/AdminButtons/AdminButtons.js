import React from "react";
import { Link } from "react-router-dom";
import { Button, Popover, OverlayTrigger } from "react-bootstrap";

/** Admin only buttons component
 *
 * renders an edit and delete button for these compononents:
 * - RoundDetails
 * - TournamentDetails
 * - CourseDetails
 *
 * Props needed:
 *  -updatePath: the path to the edit page
 *  -handleDelete: function to handle the deletion of a round,
 *    tournament, or course
 *
 *  Note: delete button is nested inside popover for extra caution
 */

const AdminButtons = ({ updatePath, handleDelete, light }) => {
  console.debug(
    "EditDeleteBtns",
    "updatePath=",
    updatePath,
    "handleDelete=",
    handleDelete
  );

  //popover stuff
  // const [popoverOpen, setPopoverOpen] = useState(false);
  // const toggle = () => setPopoverOpen(!popoverOpen);

  //grab item for deletion warning message
  const item = updatePath.split("/")[1].slice(0, -1);

  const popover = (
    <Popover id="popover-basic">
      <Popover.Header className="bg-warning text-center">
        WARNING!
      </Popover.Header>
      <Popover.Body>
        <p>
          Are you sure you want to delete this {item}? This action cannot be
          undone!
        </p>
        <Button variant="danger" className="mb-5 btn-sm" onClick={handleDelete}>
          Confirm
        </Button>
      </Popover.Body>
    </Popover>
  );

  return (
    <div className="row justify-content-center">
      <div className="col-auto">
        <div>
          <OverlayTrigger trigger="click" placement="bottom" overlay={popover}>
            <Button
              variant={light ? "outline-light" : "outline-danger"}
              id="Popover1"
              type="button"
              className="btn-sm"
            >
              Delete
            </Button>
          </OverlayTrigger>
        </div>
      </div>
      <div className="col-auto">
        <Link to={updatePath}>
          <Button
            variant={light ? "outline-light" : "outline-primary"}
            className="px-3 btn-sm"
          >
            Edit
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default AdminButtons;
