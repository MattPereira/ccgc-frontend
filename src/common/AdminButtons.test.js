import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import AdminButtons from "./AdminButtons";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <AdminButtons updatePath="/courses/pebble-beach/edit" />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <AdminButtons updatePath="/courses/pebble-beach/edit" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays", function () {
  render(
    <MemoryRouter>
      <AdminButtons updatePath="/courses/pebble-beach/edit" />
    </MemoryRouter>
  );

  const editBtn = screen.getByText("Edit");
  expect(editBtn).toBeInTheDocument();

  const deleteBtn = screen.getByText("Delete");
  expect(deleteBtn).toBeInTheDocument();
});
