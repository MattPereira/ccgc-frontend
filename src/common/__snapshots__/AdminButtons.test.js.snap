// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center mb-3"
  >
    <div
      class="col-auto"
    >
      <div>
        <button
          class="btn-sm btn btn-outline-danger"
          id="Popover1"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <div
      class="col-auto"
    >
      <a
        href="/courses/pebble-beach/edit"
      >
        <button
          class="px-3 btn-sm btn btn-outline-primary"
          type="button"
        >
          Edit
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
